swarm/src/lib.rs

```diff
@@ -51,7 +51,7 @@
 //! The [`ConnectionHandler`] trait defines how each active connection to a
 //! remote should behave: how to handle incoming substreams, which protocols
 //! are supported, when to open a new outbound substream, etc.
-//!
+#![allow(rustdoc::private_intra_doc_links)]
```

Comment:
I am not sure if it makes sense to allow this. Do the offending links work on `docs.rs`? Because if not, then there isn't really much of a point I think.

I'd be in favor of rewriting the docs instead.

@mxinden Thoughts?

Reply:
> I'd be in favor of rewriting the docs instead.

I would be in favor of this as well.

Reply:
The links show up in `docs.rs`, though the links do not work because the `connection` and `connection/pool` modules are private, so completely agree on the doc rewrite (assuming making `connection` and `connection/pool` public isn't a good option).  Links to the offending documentation:

* https://github.com/libp2p/rust-libp2p/blob/master/swarm/src/dial_opts.rs#L124-L129
* https://github.com/libp2p/rust-libp2p/blob/master/swarm/src/dial_opts.rs#L198-L203
* https://github.com/libp2p/rust-libp2p/blob/master/swarm/src/lib.rs#L1371-L1377

I can lift the doc comments from the referenced functions and replace the private links with them, though this is not DRY, so it adds to the doc-update burden.  Thoughts @thomaseizinger ?

Reply:
I'd suggest we simply remove the docs from the non-public items and put them on the public ones. In other words, inline the documentation we are currently referring to and deleting it from there!

Reply:
This is done, though I left the documentation in the private functions alone as it looked out of place with the other items documented.  It may be out of scope for this PR, but it's worth considering changing `///` to `//` documentation in private items.